### PR TITLE
Make mypy more strict, add all missing type annotations

### PR DESCRIPTION
**Pull request by msurdi**

### CODE REVIEW

File: failfast/failfast.py [hunk 1]
```diff
@@ -13,32 +14,31 @@ class FailfastException(Exception):
 
 def failfast(name: str,
              timeout_seconds: int = 300,
-             store: Optional[Store] = None,
```
Comment (Kartones) [inline failfast/failfast.py]: Why did you remove the `Optional`, [it is correct](http://mypy.readthedocs.io/en/stable/kinds_of_types.html?highlight=Optional#the-type-of-none-and-optional-types) for default value `None`...

Comment (msurdi) [reply failfast/failfast.py]: It's redundant, as mypy is already inferring it is Optional given it has been assigned None as default value and we are not using --strict-optional command line flag (see [here](http://mypy.readthedocs.io/en/latest/kinds_of_types.html#experimental-strict-optional-type-and-none-checking)). 

I can add it back for "documentation" purposes, but not for safety as it is already detected as Optional correctly

Comment (Kartones) [reply failfast/failfast.py]: Yep, I was referring more about documentation purposes, it's true that mypy works nicely and detects if you forgive to set it but to me reads weird to se something with a default `None` not being either optional or a union with none...

But is just personal taste, it is true that with the strict option we can assume all those implicit optionals

Comment (msurdi) [reply failfast/failfast.py]: I've added it back as I think it makes it clearer, given I'm using Optional in other arguments this was quite weird to not use it here too, good catch.
